### ISSUE TITLE
[Minor Bug] - having a DPI above 100% causes false negatives for window visibility check

### ISSUE THREAD

**Issue by ataboo** [ciribob/DCS-SimpleRadioStandalone#312]
I run my main monitor at 3840 x 2160 and a 1920 x 1080 monitor on each side.  The main monitor has its DPI set to 150%.  If move the SRS client to either secondary monitor and close it, it shows "The SRS client window is no longer visible..." message on next startup.   Here's an example log:

> 2019-01-14 18:56:02.7127 | Ciribob.DCS.SimpleRadio.Standalone.Client.UI.MainWindow | Checking window visibility for main client window {X=-686,Y=1402} 
> 2019-01-14 18:56:02.7439 | Ciribob.DCS.SimpleRadio.Standalone.Client.UI.MainWindow | Checking window visibility for radio overlay {X=300,Y=300} 
> 2019-01-14 18:56:02.7439 | Ciribob.DCS.SimpleRadio.Standalone.Client.UI.MainWindow | Checking window visibility for AWACS overlay {X=300,Y=300} 
> 2019-01-14 18:56:02.7439 | Ciribob.DCS.SimpleRadio.Standalone.Client.UI.MainWindow | Checking screen \\.\DISPLAY1 with bounds {X=5760,Y=1604,Width=2880,Height=1620} for window visibility 
> 2019-01-14 18:56:02.7596 | Ciribob.DCS.SimpleRadio.Standalone.Client.UI.MainWindow | Checking screen \\.\DISPLAY2 with bounds {X=-2880,Y=1604,Width=2880,Height=1620} for window visibility 
> 2019-01-14 18:56:02.7596 | Ciribob.DCS.SimpleRadio.Standalone.Client.UI.MainWindow | Checking primary screen \\.\DISPLAY3 with bounds {X=0,Y=0,Width=3840,Height=2160} for window visibility 
> 2019-01-14 18:56:02.7596 | Ciribob.DCS.SimpleRadio.Standalone.Client.UI.MainWindow | Radio overlay {X=300,Y=300} is visible on primary screen \\.\DISPLAY3 with bounds {X=0,Y=0,Width=3840,Height=2160} 
> 2019-01-14 18:56:02.7596 | Ciribob.DCS.SimpleRadio.Standalone.Client.UI.MainWindow | AWACS overlay {X=300,Y=300} is visible on primary screen \\.\DISPLAY3 with bounds {X=0,Y=0,Width=3840,Height=2160} 

Also after it has reset the app position, the favorites are not listed and can not be added or removed until app reload.

If I set the monitor to 100% DPI it works fine (have to log out and in for changes to effect).

I'm not super familiar with dotnet so I only dug a bit but there was mention of [declaring the app DPI aware](https://docs.microsoft.com/en-us/dotnet/framework/winforms/high-dpi-support-in-windows-forms) to get the proper dimensions from `System.Windows.Forms.Screen`.  I have no idea how deep that rabbit hole goes though so just thought I'd mention it.

**Comment by MorpheusXAUT** [comment]
We've encountered this issue before after introducing the window visibility check and I believe did not find a quick or easy solution back then, which is why we [introduced a config-file only flag to disable the visibility check](https://github.com/ciribob/DCS-SimpleRadioStandalone/commit/b9cc2ce9df27433d3522cccba9fee665825d2da5).
If you open the `client.cfg` in your SRS folder and change `DisableWindowVisibilityCheck=false` to `DisableWindowVisibilityCheck=true`, SRS will skip the visibility test on startup and thus not bother you with resetting the window due to the higher DPI issue. I know, it's not really a solution, but at least a workaround for now 😄 

I'll definitely look into declaring SRS as DPI aware, since I'm not sure we've tried that before, but it might not be a full solution either since it apparently requires Windows 10, while we still have some Windows 7 users as well. Will still check it out and see though, thanks for the link 👍

**Comment by realtimeprojects** [comment]
Yes, seems that windows messes up with the pixel positions when you change the scale and layout on your screen. Having set the **size of text** on win10 to 175%, x=position of the srs window is now around 1600 in the global.cfg when I put my window to the right side of the screen while it was around 3000 before scaling up the font.

(Single screen setup with 3840x2160 display resolution)

In case you have the old values of 3000 in your global.cfg and you change your scale, srs window is no longer visible.

Not having looked deeply into that, I think we have to do some magic with https://docs.microsoft.com/en-us/dotnet/api/system.windows.forms.control.logicaltodeviceunits?view=netframework-4.8 (available from .net 4.7) in order to fix that.

**Comment by realtimeprojects** [comment]
Stumbled over this:
https://stackoverflow.com/questions/52299901/find-devicepixelratio-according-to-both-page-zoom-level-and-windows-10-dpi-setti

**Comment by Pfheonix** [comment]
Is anyone looking for something which will fix this? I can try to take a crack at it if you're open to public pull requests.

**Comment by ciribob** [comment]
@Pfheonix if you can solve it it would be much appreciated!
